### PR TITLE
[JENKINS-67032] Address dead URLs

### PR DESCRIPTION
**Pull request by NotMyFault**
This PR replaces a bunch of dated URLs, such as wiki.jenkins-ci.org and replaces http with https in applicable places.
The latter addresses cases, where browsers block domains instead of redirecting, assuming one is set.

Therefore, a few links had to be updated to forward to a working destination.

### Proposed changelog entries

* Replace outdated URLs with working redirects.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: war/src/main/fonts/font-awesome/css/font-awesome.css [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*!
- *  Font Awesome 4.6.2 by @davegandy - http://fontawesome.io - @fontawesome
- *  License - http://fontawesome.io/license (Font: SIL OFL 1.1, CSS: MIT License)
+ *  Font Awesome 4.6.2 by @davegandy - https://fontawesome.com - @fontawesome
+ *  License - https://fontawesome.com/license (Font: SIL OFL 1.1, CSS: MIT License)
```
Comment (jtnord) [inline war/src/main/fonts/font-awesome/css/font-awesome.css:3]: Given this is likely a local copy of font-awesome CSS file should this be changed from the original?

File: war/src/main/fonts/font-awesome/css/font-awesome.min.css [hunk 2]
```diff
@@ -1,4 +1,4 @@
 /*!
- *  Font Awesome 4.6.2 by @davegandy - http://fontawesome.io - @fontawesome
- *  License - http://fontawesome.io/license (Font: SIL OFL 1.1, CSS: MIT License)
+ *  Font Awesome 4.6.2 by @davegandy - https://fontawesome.com - @fontawesome
```
Comment (jtnord) [inline war/src/main/fonts/font-awesome/css/font-awesome.min.css:2]: Same comment as on the non minified css

**Review by jtnord — APPROVED** [review]

File: core/src/main/resources/hudson/tools/ZipExtractionInstaller/help.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div>
     Downloads a tool archive and installs it within Jenkins's working directory.
-    Example: <code>http://apache.promopeddler.com/ant/binaries/apache-ant-1.7.1-bin.zip</code>
+    Example: <code>https://downloads.apache.org/ant/binaries/apache-ant-1.10.12-bin.zip</code>
     (or whatever mirror is closest to your server)
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/tools/ZipExtractionInstaller/help.html:4]: While the old URL was weird, now the mirror comment makes no sense anymore. Delete the part in parentheses and wipe all translations?

File: war/src/main/fonts/font-awesome/css/font-awesome.css (hunk 1 above)
Comment (daniel-beck) [reply war/src/main/fonts/font-awesome/css/font-awesome.css:3]: Folks have been submitting typo fixes in our YUI debug files for a looooong time and they've all been approved...

File: war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel.html [hunk 4]
```diff
@@ -3,8 +3,8 @@
     to establish a communication channel.
     But some security sensitive network can prevent you from making this connection.
     This can also happen when Jenkins runs behind a load balancer,
-    <a href="http://www.serverwatch.com/tutorials/article.php/3290851">apache reverse proxy</a>
-    into <a href="http://en.wikipedia.org/wiki/Demilitarized_zone_(computing)">DMZ</a>, and so on.
+    <a href="https://www.serverwatch.com/guides/reverse-proxying-with-apache-2-0/">apache reverse proxy</a>
```
Comment (daniel-beck) [inline war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel.html:6]: Probably better to link to our own docs?

```suggestion
    <a href="https://www.jenkins.io/doc/book/system-administration/reverse-proxy-configuration-apache/">apache reverse proxy</a>
```

File: war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_de.html [hunk 5]
```diff
@@ -2,8 +2,8 @@
     Wird ein Agent über JNLP gestartet, versucht dieser über einen bestimmten TCP-Port
     den Jenkins-Master zu erreichen. In abgesicherten Netzwerken kann dieser Verbindungsaufbau
     jedoch scheitern. Probleme können auch dadurch entstehen, wenn Jenkins hinter einem Lastverteiler (load balancer),
-    <a href="http://www.serverwatch.com/tutorials/article.php/3290851">einem Apache Reverse-Proxy</a>
-    in eine <a href="http://en.wikipedia.org/wiki/Demilitarized_zone_(computing)">DMZ</a> o.ä. 
+    <a href="https://www.serverwatch.com/guides/reverse-proxying-with-apache-2-0/">einem Apache Reverse-Proxy</a>
```
Comment (daniel-beck) [inline war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_de.html:5]:
```suggestion
    <a href="https://www.jenkins.io/doc/book/system-administration/reverse-proxy-configuration-apache/">einem Apache Reverse-Proxy</a>
```

File: war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_fr.html [hunk 6]
```diff
@@ -3,8 +3,8 @@
     TCP spécifique, afin d'établir un canal de communication.
     Certains réseaux très sécurisés peuvent vous empècher d'effectuer cette connexion.
     Cela peut également être le cas quand Jenkins tourne derrière un répartiteur de charge (load balancer),
-    un <a href="http://www.serverwatch.com/tutorials/article.php/3290851">reverse proxy apache</a>
-    dans une <a href="http://en.wikipedia.org/wiki/Demilitarized_zone_(computing)">zone démilitarisée (DMZ)</a>, etc.
+    un <a href="https://www.serverwatch.com/guides/reverse-proxying-with-apache-2-0/">reverse proxy apache</a>
```
Comment (daniel-beck) [inline war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_fr.html:6]:
```suggestion
    un <a href="https://www.jenkins.io/doc/book/system-administration/reverse-proxy-configuration-apache/">reverse proxy apache</a>
```

File: war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_it.html [hunk 7]
```diff
@@ -3,8 +3,8 @@
     Jenkins per stabilire un canale di comunicazione.
     Alcune reti configurate in modo sicuro, tuttavia, possono impedire l'instaurazione di questa connessione.
     Ciò può succedere anche quando Jenkins è in esecuzione mascherato da un bilanciatore di carico,
-    <a href="http://www.serverwatch.com/tutorials/article.php/3290851">da un proxy inverso Apache</a>
-    è in una <a href="http://en.wikipedia.org/wiki/Demilitarized_zone_(computing)">DMZ</a>, e così via.
+    <a href="https://www.serverwatch.com/guides/reverse-proxying-with-apache-2-0/">da un proxy inverso Apache</a>
```
Comment (daniel-beck) [inline war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_it.html:6]:
```suggestion
    <a href="https://www.jenkins.io/doc/book/system-administration/reverse-proxy-configuration-apache/">da un proxy inverso Apache</a>
```

File: war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_ja.html [hunk 8]
```diff
@@ -2,8 +2,8 @@
     エージェントがJNLPで起動するとき、エージェントは通信チャネルを確立するために、Jenkinsのある特定のTCPポートに接続しようとします。
     しかし、セキュリティに神経質なネットワークでは接続できないことがあります。
     これは、Jenkinsがロードバランサーや
-    <a href="http://ja.wikipedia.org/wiki/%E9%9D%9E%E6%AD%A6%E8%A3%85%E5%9C%B0%E5%B8%AF_(%E3%82%B3%E3%83%B3%E3%83%94%E3%83%A5%E3%83%BC%E3%82%BF%E3%82%BB%E3%82%AD%E3%83%A5%E3%83%AA%E3%83%86%E3%82%A3)">DMZ</a>への
-    <a href="http://www.serverwatch.com/tutorials/article.php/3290851">Apacheリバースプロクシー</a>の背後にある場合などにも生じます。
+    <a href="https://ja.wikipedia.org/wiki/%E9%9D%9E%E6%AD%A6%E8%A3%85%E5%9C%B0%E5%B8%AF_(%E3%82%B3%E3%83%B3%E3%83%94%E3%83%A5%E3%83%BC%E3%82%BF%E3%82%BB%E3%82%AD%E3%83%A5%E3%83%AA%E3%83%86%E3%82%A3)">DMZ</a>への
+    <a href="https://www.serverwatch.com/guides/reverse-proxying-with-apache-2-0/">Apacheリバースプロクシー</a>の背後にある場合などにも生じます。
```
Comment (daniel-beck) [inline war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_ja.html:6]:
```suggestion
    <a href="https://www.jenkins.io/doc/book/system-administration/reverse-proxy-configuration-apache/">Apacheリバースプロクシー</a>の背後にある場合などにも生じます。
```

File: war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_zh_TW.html [hunk 9]
```diff
@@ -2,8 +2,8 @@
     Agent 透過 JNLP 啟動時，其代理程式會試著連回 Jenkins 特定的 TCP 連接埠，建立通訊通道。
     但是某些安全敏感的網路可能不能讓您這樣建立連線。
     Jenkins 在負載平衡器、
-    <a href="http://www.serverwatch.com/tutorials/article.php/3290851">Apache 反向代理</a>到
-    <a href="http://en.wikipedia.org/wiki/Demilitarized_zone_(computing)">DMZ</a> 等環境中也可能無法建立連線。
+    <a href="https://www.serverwatch.com/guides/reverse-proxying-with-apache-2-0/">Apache 反向代理</a>到
```
Comment (daniel-beck) [inline war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel_zh_TW.html:5]:
```suggestion
    <a href="https://www.jenkins.io/doc/book/system-administration/reverse-proxy-configuration-apache/">Apache 反向代理</a>到
```

**Review by daniel-beck — APPROVED** [review]
Thanks!

**Review by MarkEWaite — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
All updated links look fine. Some of them are user-facing, so IMHO we should handle it as bug. Maybe also a candidate for backporting, but we would need a Jira issue for it

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by NotMyFault** [comment]
> All updated links look fine. Some of them are user-facing, so IMHO we should handle it as bug. Maybe also a candidate for backporting, but we would need a Jira issue for it

I created an [issue](https://issues.jenkins.io/browse/JENKINS-67032) in case that may be a candidate for backporting.

**Comment by daniel-beck** [comment]
> We may merge it in 24 hours if there is no negative feedback. Please see the merge process documentation for more information about the merge process

Please don't, this will just make the Thursday integration of 2.319 more annoying.

**Pull request merged**